artifactory/system.go

```diff
@@ -33,6 +40,1112 @@ func (v Versions) String() string {
+type MailServer struct {
+	Enabled        *bool   `yaml:"enabled,omitempty" xml:"enabled,omitempty"`
+	ArtifactoryUrl *string `yaml:"artifactoryUrl,omitempty" xml:"artifactoryUrl,omitempty"`
+	From           *string `yaml:"from,omitempty" xml:"from,omitempty"`
+	Host           *string `yaml:"host,omitempty" xml:"host,omitempty"`
+	Username       *string `yaml:"username,omitempty" xml:"username,omitempty"`
+	Password       *string `yaml:"password,omitempty" xml:"password,omitempty"`
+	Port           *int    `yaml:"port,omitempty" xml:"port,omitempty"`
+	SubjectPrefix  *string `yaml:"subjectPrefix,omitempty" xml:"subjectPrefix,omitempty"`
+	Ssl            *bool   `yaml:"ssl,omitempty" xml:"ssl,omitempty"`
+	Tls            *bool   `yaml:"tls,omitempty" xml:"tls,omitempty"`
+	Reset          *bool   `yaml:"-" xml:"-"`
+}
+
+// MarshalYAML implements the Marshaller interface.
+func (m MailServer) MarshalYAML() (interface{}, error) {
+	if m.Reset != nil && *m.Reset {
+		return nil, nil
+	}
+	return m, nil
+}
+
+// XrayConfig represents Xray related settings in Artifactory's Configuration
+// Descriptor. This is undocumented in YAML Configuration File.
+//
+type XrayConfig struct {
+	Enabled                       *bool   `yaml:"enabled,omitempty" xml:"enabled,omitempty"`
+	BaseUrl                       *string `yaml:"baseUrl,omitempty" xml:"baseUrl,omitempty"`
+	User                          *string `yaml:"user,omitempty" xml:"user,omitempty"`
+	Password                      *string `yaml:"password,omitempty" xml:"password,omitempty"`
+	ArtifactoryId                 *string `yaml:"artifactoryId,omitempty" xml:"artifactoryId,omitempty"`
+	XrayId                        *string `yaml:"xrayId,omitempty" xml:"xrayId,omitempty"`
+	AllowDownloadsXrayUnavailable *bool   `yaml:"allowDownloadsXrayUnavailable,omitempty" xml:"allowDownloadsXrayUnavailable,omitempty"`
+	AllowBlockedArtifactsDownload *bool   `yaml:"allowBlockedArtifactsDownload,omitempty" xml:"allowBlockedArtifactsDownload,omitempty"`
+	BypassDefaultProxy            *bool   `yaml:"bypassDefaultProxy,omitempty" xml:"bypassDefaultProxy,omitempty"`
+	Proxy                         *string `yaml:"proxy,omitempty" xml:"proxy,omitempty"`
+}
+
+// BintrayConfig represents Bintray settings in Artifactory's Configuration
+// Descriptor. This is undocumented in YAML Configuration File.
+//
+type BintrayConfig struct {
+	UserName        *string `yaml:"userName,omitempty" xml:"userName,omitempty"`
+	ApiKey          *string `yaml:"apiKey,omitempty" xml:"apiKey,omitempty"`
+	FileUploadLimit *int    `yaml:"fileUploadLimit,omitempty" xml:"fileUploadLimit,omitempty"`
+}
+
+// Proxy represents a Proxy setting in Artifactory's Global Configuration Descriptor.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/YAML+Configuration+File#YAMLConfigurationFile-General(General,PropertySets,Proxy,Mail)
+type Proxy struct {
+	Key             *string `yaml:"-" xml:"key,omitempty"`
```

Comment:
I noticed we're not allowing this field to be serialized in YAML due to the `-` tag.

This peaked my curiosity so was this intentional?

Reply:
The YAML format is map[key]config so the custom YAML marshaler implemented on type Proxies transforms the slice of Proxies to this format.